### DISCUSSION TITLE
Usage guidelines in Class Components? HOC?

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#3362] [Q&A] [answered]
Can you use Chakra components like [Modal](https://chakra-ui.com/docs/overlay/modal) inside **class components**?

1. How would you map the `useDisclosure()` hook to use natively without HOC?
2. If you have to use HOC is this the right way of doing it?

**Comment by dodas** [comment, marked as answer]
Hey @bachmannn!

The `useDisclosure` hook just returns a piece of state - the `isOpen` boolean, and 3 functions to update it: `onOpen`, `onClose` and `onToggle`. You really don't have to use `useDisclosure` at all. All you need is a single boolean in component state and functions to update it:

```jsx
export default class FooClass extends Component<FooProps, FooState> {
  state = {
    isModalOpen: false,
  }

  openModal = () => {
    this.setState({ isModalOpen: true })
  }

  closeModal = () => {
    this.setState({ isModalOpen: false })
  }

  render() {
    return (
      <>
        <Button onClick={this.openModal}>Open Modal</Button>
        <Modal isOpen={this.state.isModalOpen} onClose={this.closeModal}>
          // ...
        </Modal>
      </>
    )
  }
}
```

You could still make a HOC for it, but I don't feel like that's necessary.
Does this clarify it for you?
